utils/project/sync_test.go

```diff
@@ -12,70 +12,188 @@
 package project
 
 import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 )
 
+type noIgnoredPaths struct {
+	Field1 []string `json:"field1"`
+	Field2 string   `json:"field2"`
+}
+
+var testFolder string
+var cwSettingsPopulatedPath string
+var cwSettingsEmptyPath string
+var cwSettingsNoIgnoredPaths string
+
+func TestMain(m *testing.M) {
+	setup()
+	code := m.Run()
+	shutdown()
+	os.Exit(code)
+}
+
+func setup() {
+	// Create directories to write .cw-settings files to
+	testFolder = "sync_test_folder_delete_me"
+	cwSettingsPopulatedPath = testFolder + "/cwSettingsPopulated"
+	cwSettingsEmptyPath = testFolder + "/cwSettingsEmpty"
+	cwSettingsNoIgnoredPaths = testFolder + "/cwSettingsNoIgnoredPaths"
+	populatedIgnoredPaths := CWSettings{
+		IgnoredPaths: []string{
+			"testfile",
+			"anothertestfile",
+		},
+	}
+	emptyIgnoredPaths := CWSettings{
+		IgnoredPaths: []string{},
+	}
+	noIgnoredPaths := noIgnoredPaths{
+		Field1: []string{"Something", "Else"},
+		Field2: "Something Else",
+	}
+
+	// Create test directory to store .cw-settings files in
+	os.Mkdir(testFolder, 0777)
+	os.Mkdir(cwSettingsPopulatedPath, 0777)
+	os.Mkdir(cwSettingsEmptyPath, 0777)
+	os.Mkdir(cwSettingsNoIgnoredPaths, 0777)
+
+	// Create .cw-settings files
+	file, _ := json.Marshal(populatedIgnoredPaths)
+	ioutil.WriteFile(cwSettingsPopulatedPath+"/.cw-settings", file, 0644)
+	file, _ = json.Marshal(emptyIgnoredPaths)
+	ioutil.WriteFile(cwSettingsEmptyPath+"/.cw-settings", file, 0644)
+	file, _ = json.Marshal(noIgnoredPaths)
+	ioutil.WriteFile(cwSettingsNoIgnoredPaths+"/.cw-settings", file, 0644)
+}
+
+func shutdown() {
```

Comment:
this func name doesnt tell me it is removing a test folder. I would suggest using `removeTestDir()` or something similar

Reply:
have removed this function